pkg/agent/openflow/pipeline.go

```diff
@@ -602,26 +607,49 @@ func (f *featurePodConnectivity) podClassifierFlow(podOFPort uint32, isAntreaFle
 
 // podUplinkClassifierFlows generates the flows to mark the packets with target destination MAC address from uplink/bridge
 // port, which are needed when uplink is connected to OVS bridge and Antrea IPAM is configured.
-func (f *featurePodConnectivity) podUplinkClassifierFlows(dstMAC net.HardwareAddr) []binding.Flow {
+func (f *featurePodConnectivity) podUplinkClassifierFlows(dstMAC net.HardwareAddr, vlanID uint16) []binding.Flow {
 	cookieID := f.cookieAllocator.Request(f.category).Raw()
-	return []binding.Flow{
-		// This generates the flow to mark the packets from uplink port.
-		ClassifierTable.ofTable.BuildFlow(priorityHigh).
-			Cookie(cookieID).
-			MatchInPort(config.UplinkOFPort).
-			MatchDstMAC(dstMAC).
-			Action().LoadRegMark(FromUplinkRegMark).
-			Action().GotoStage(stageConntrackState).
-			Done(),
-		// This generates the flow to mark the packets from bridge local port.
-		ClassifierTable.ofTable.BuildFlow(priorityHigh).
-			Cookie(cookieID).
-			MatchInPort(config.BridgeOFPort).
-			MatchDstMAC(dstMAC).
-			Action().LoadRegMark(FromBridgeRegMark).
-			Action().GotoStage(stageConntrackState).
-			Done(),
+	var flows []binding.Flow
+	nonVLAN := true
+	if vlanID > 0 {
+		nonVLAN = false
+	}
+	for _, ipProtocol := range f.ipProtocols {
+		ctZoneTypeRegMark := IPCtZoneTypeRegMark
+		if ipProtocol == binding.ProtocolIPv6 {
+			ctZoneTypeRegMark = IPv6CtZoneTypeRegMark
+		}
+		flows = append(flows,
+			// This generates the flow to mark the packets from uplink port.
+			ClassifierTable.ofTable.BuildFlow(priorityNormal).
+				Cookie(cookieID).
+				MatchInPort(config.UplinkOFPort).
+				MatchDstMAC(dstMAC).
+				MatchVLAN(nonVLAN, vlanID, nil).
```

Comment:
Is nonVLAN redundant with vlanID?

Reply:
No, if nonVLAN is true, vlanID and mask will be ignored.
This call support two cases:
nonVLAN=true and vlanID = 0
nonVLAN=false and vlanID > 0

Reply:
This is to make code tidy.